src/helpers/composeVersionChecker.ts

```diff
@@ -0,0 +1,103 @@
+import compareVersions from 'compare-versions'
+
+interface IVersionMap {
+  [key: string]: string
+}
+
+// Here the engine version is the lowest needed to support all docker compose features
+const composeVersionMap: IVersionMap = {
```

Comment:
use a JSON file to store these mappings and enable `resolveJsonModule` typescript option. (may require `esModuleInterop` and/or `allowSyntheticDefaultImports`